core/state_processor.go

```diff
@@ -102,6 +108,13 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 }
 
 func applyTransaction(msg *Message, config *params.ChainConfig, gp *GasPool, statedb *state.StateDB, blockNumber *big.Int, blockHash common.Hash, tx *types.Transaction, usedGas *uint64, evm *vm.EVM) (*types.Receipt, error) {
+	var receipt *types.Receipt
+	if evm.Config.Tracer != nil {
+		evm.Config.Tracer.CaptureTxStart(evm, tx)
+		defer func() {
+			evm.Config.Tracer.CaptureTxEnd(receipt)
```

Comment:
We've previously discussed and decided against this (https://github.com/ethereum/go-ethereum/issues/25953). But now that we're making a bigger change to the interface is a good chance to update this behavior too.

I think it's valuable to instead of simply dropping the invalid call/create capture it with an error and let users know why the scope was not entered. The same for tx validation checks (pre checks).

I've applied a commit with both of these changes. This naturally broke some changes in the call tracer because there are now extra call frames which failed validation. Those are fixed too. I'll add a backwards-compatibility note in the PR description.